src/utils/message-formatter.js

```diff
@@ -0,0 +1,83 @@
+/**
+  @module with utilities for chain links
+**/
+
+/**
+  @function packMessage
+  Convert the given parameters into a frozen (@see Object.freeze()) message package
+  @param level {string} - a message log level (@see ChainLink @class for additional info)
+  @param msg {Object|Error} - a message payload. Either a text string or an Error object
+  @param metas {Object} - metadata
+  @return {Object} - frozen message package object
+**/
+function packMessage(level, msg, ...metas) {
+  level = level || process.env.DEFAULT_LOG_LEVEL || 'info'; //eslint-disable-line
+  // if plain text
+  const message = {
+    level,
+    text: msg || '',
+    meta: {
+      instanceId: process.env.HOSTNAME,
+      notify: true
+    }
+  };
+
+  // if error
+  if (msg instanceof Error) {
+    Object.assign(message, {
+      text: msg.message || 'Error: ',
+      meta: {
+        stack: msg.stack,
+        notify: true,
+        instanceId: process.env.HOSTNAME
+      }
+    });
+  }
+  // all metas are included as message meta
+  if (metas.length > 0) {
+    const metaData = metas.reduce((sum, next) => Object.assign({}, sum, next));
+    Object.assign(message.meta, metaData);
+  }
+  return Object.freeze(message);
```

Comment:
since message can be complex, you should use deepFreeze module to freeze any sub objects.